src/nix/modules/service/image.nix

```diff
@@ -9,7 +9,12 @@ let
     (pkgs.writeText "dummy-config.json" (builtins.toJSON config.image.rawConfig))
   ];
 
-  builtImage = pkgs.dockerTools.buildLayeredImage {
+  buildOrStreamLayeredImage = args:
+    if pkgs.dockerTools?streamLayeredImage
+    then pkgs.dockerTools.streamLayeredImage args // { isExe = true; }
```

Comment:
Technically someone could override `build.image` causing a `streamLayeredImage` without `isExe = true`, but that will be fixed by https://github.com/NixOS/nixpkgs/pull/99375